src/main/kotlin/it/pagopa/checkout/authservice/client/oneidentity/OneIdentityClient.kt

```diff
@@ -0,0 +1,39 @@
+package it.pagopa.checkout.authservice.client.oneidentity
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.util.*
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.web.util.UriComponentsBuilder
+
+@Component
+class OneIdentityClient(
+    @Value("\${oneidentity.base-url}") private val oneIdentityBaseUrl: String,
+    @Value("\${oneidentity.redirect-uri}") private val redirectUri: String,
+    @Value("\${oneidentity.client-id}") private val clientId: String,
+) {
+    fun buildLoginUrl(): String {
+
+        // Value opaque to the server, used by the client to track its session.
+        // It will be returned as received.
+        val state = UUID.randomUUID().toString()
+
+        // Represents a cryptographically strong random string that is used to prevent
+        // intercepted responses from being reused.
+        val nonce = UUID.randomUUID().toString()
+
+        val encodedUrl = URLEncoder.encode(redirectUri, StandardCharsets.UTF_8.toString())
+
+        return UriComponentsBuilder.fromUriString(oneIdentityBaseUrl)
+            .path("/login")
+            .queryParam("response_type", "code")
+            .queryParam("scope", "openid")
+            .queryParam("client_id", clientId)
+            .queryParam("state", state)
+            .queryParam("nonce", nonce)
+            .queryParam("redirect_uri", encodedUrl)
+            .build()
+            .toUriString()
+    }
```

Comment:
```suggestion
return Mono
            .just(oneIdentityBaseUrl)
            .map {UriComponentsBuilder.fromUriString(it)
                .path("/login")
                .queryParam("response_type", "code")
                .queryParam("scope", "openid")
                .queryParam("client_id", clientId)
                .queryParam("state", state)
                .queryParam("nonce", nonce)
                .queryParam("redirect_uri", encodedUrl)
                .build()
                .toUriString()
            }
    }
```

rewrite this function in a reactive way so that it returns a Mono, leaving the error propagation up to the reactive framework (an exception raised in a map methods will result in a mono error to be produced)

Reply:
Good suggestion, addressed with [5147de5](https://github.com/pagopa/pagopa-checkout-auth-service/pull/12/commits/5147de508232d1618d5ea358037f48e22aeb8778) ✅